### PR TITLE
Post forms

### PR DESCRIPTION
**Pull request by betulerkoc**
- Create receiver form and donator form components

- Create form page and render components with toggle button

-Add data to firebase

form-page

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from "react";
 import Home from "./containers/Home";
 import Footer from "./components/Footer";
+import Form from "./containers/Form";
```
Comment (Ammar-64) [inline src/App.jsx]: Can you give the form component a more meaningful name, i.e. DonateFrom maybe or whatever you find more convenient

File: src/components/DonatorForm/index.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    <div>
+      <Container>
+        <Form onSubmit={addDonatorPost}>
+          <Form.Group>
+            <Form.Label>Donation Type</Form.Label>
+            <Form.Control
+              as="select"
+              onChange={(e) => handlePostChange(e, "type")}
+            >
+              <option value="technology">Technology</option>
```
Comment (Ammar-64) [inline src/components/DonatorForm/index.jsx]: Can you use i18next for all the text to reduce the amount of work we have in the final couple of days

File: src/components/ReceiverForm/index.jsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import React, { useState } from "react";
+import { Container, Form, Button, Col, Row } from "react-bootstrap";
+import db from "../../firebaseConfig";
+import "./index.css";
+
```
Comment (Ammar-64) [inline src/components/ReceiverForm/index.jsx:6]: The receiver and donator forms look similar to me, instead you can do only one form, and change what’s being posted to firebase according to the props coming to the component, you don’t need to repeat yourself

File: src/containers/Form/index.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+              value={radio.value}
+              checked={radioValue === radio.value}
+              onChange={(e) => setRadioValue(e.currentTarget.value)}
+            >
+              {radio.name}
+            </ToggleButton>
+          ))}
+        </ButtonGroup>
+      </Row>
+      {radioValue === "1" ? <ReceiverForm /> : <DonatorForm />}
```
Comment (Ammar-64) [inline src/containers/Form/index.jsx]: Here is where you can pass props to differentiate donator from receiver in the same form

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
Solid work but can you take the comments I adeed into consideration

File: src/components/DonatorForm/index.jsx (hunk 2 above)
Comment (betulerkoc) [reply src/components/DonatorForm/index.jsx]: PTAL

File: src/App.jsx (hunk 1 above)
Comment (betulerkoc) [reply src/App.jsx]: PTAL

**Review by olaouso — APPROVED** [review]

File: src/components/DonatorForm/index.jsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  const handlePostChange = (e, key) => {
+    console.log(e.target.value);
+    setDonatorPostData({ ...donatorPostData, [key]: e.target.value });
+    console.log(donatorPostData);
+  };
+  const { t } = useTranslation();
+
+  const addDonatorPost = (e) => {
+    e.preventDefault();
+    console.log("yeeyyyyyy");
```
Comment (Ammar-64) [inline src/components/DonatorForm/index.jsx]: Can you remove the console.log from here

File: src/components/ReceiverForm/index.jsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    console.log(e.target.value);
+    setReceiverPostData({ ...receiverPostData, [key]: e.target.value });
+    console.log(receiverPostData);
+  };
+
+  const { t } = useTranslation();
+
+  const addReceiverPost = (e) => {
+    e.preventDefault();
+    console.log("yeey");
```
Comment (Ammar-64) [inline src/components/ReceiverForm/index.jsx]: This console.log also

File: src/containers/NewPostForm/index.css [hunk 7]
```diff
@@ -0,0 +1,15 @@
+.row {
```
Comment (Ammar-64) [inline src/containers/NewPostForm/index.css]: I would avoid these names like row and container, they might be used bu bootstrap or someone else and make a conflict, I would go with more unique names

File: src/containers/NewPostForm/index.jsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import { useTranslation } from "react-i18next";
+import "./index.css";
+
+export default function Form() {
+  const [radioValue, setRadioValue] = useState("1");
+
+  const { t } = useTranslation();
+
+  const radios = [
+    { name: t("formsection.Receive"), value: "1" },
```
Comment (Ammar-64) [inline src/containers/NewPostForm/index.jsx]: You can also give more meaningful values here rather than 1 and 2, to make it clearer and more readable

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
Solid but please take the comments into consideration and pull and merge dev, because there’s a conflict after Ola’s code is merged

File: src/containers/NewPostForm/index.jsx (hunk 8 above)
Comment (betulerkoc) [reply src/containers/NewPostForm/index.jsx]: PTAL

File: src/containers/NewPostForm/index.css (hunk 7 above)
Comment (betulerkoc) [reply src/containers/NewPostForm/index.css]: PTAL

File: src/components/ReceiverForm/index.jsx (hunk 6 above)
Comment (betulerkoc) [reply src/components/ReceiverForm/index.jsx]: PTAL

File: src/components/DonatorForm/index.jsx (hunk 5 above)
Comment (betulerkoc) [reply src/components/DonatorForm/index.jsx]: PTAL

**Review by Ammar-64 — APPROVED** [review]
LGTM

**Pull request merged**